### DISCUSSION TITLE
Connect to local PLEX library would be nice

### DISCUSSION THREAD

**Discussion by tonblom** [Purfview/IMDb-Scout-Mod#61] [Ideas (implemented)]
Connect to local PLEX library would be nice

**Comment by Purfview** [comment]
What you meant by "connect"? What it would do?

**Reply by tonblom** [reply]
It would check if the movie is already in the plex library

**Reply by Purfview** [reply]
I just quick looked at https://support.plex.tv/articles/201638786-plex-media-server-url-commands/
I don't see it storing IMDb ID. Searching by title would be somewhat wonky. 

I'll check what can be done (no ETA).

**Reply by tonblom** [reply]
Thanks

**Comment by Purfview** [comment]
Plex indicator added in v13.0. Check if it works.
IMDb/TVDb IDs must be present in response, I could find them only in legacy agent, if you know other methods how to get them, then let me know.

**Reply by tonblom** [reply]
![image](https://user-images.githubusercontent.com/65295151/125344308-6b7dd280-e357-11eb-8f24-d86a8e84bd6c.png)

Green logo starts http://local:32400/web/index.html#!/
&
Red logo/not in Library starts http://local:32400/search?query=The%20Matrix%204&X-Plex-Token=1P----1uSx

Looks great sofar

**Reply by tonblom** [reply]
Also this one result in a 400 Bad Request:
http://local:32400/search?query=Willy%20Wonka%20&%20the%20Chocolate%20Factory&X-Plex-Token=1PqV---------Sx

Icon is red, but naming on Plex and IMDB is identical

**Reply by Purfview** [reply]
Ampersand bug should be be fixed with https://github.com/Purfview/IMDb-Scout-Mod/commit/da325c2a0a400a89cf5309dd7efa30294957b980

Those "starts" are intentional, do you expect something else to open?
For 'red' it's good for debugging, for "not found" it's good too I think.
For "found" I didn't check if exact title can be open.

**Reply by Purfview** [reply]
Btw, why that image looks so ugly?

**Comment by mastercho** [comment]
Wait if there is plex integration then why not for Emby/Jellyfin ?

**Reply by Purfview** [reply]
> Wait if there is plex integration then why not for Emby/Jellyfin ?

Support for Jellyfin is already  implemented in the same update.
Is there a point for Emby, as Jellyfin is open and free fork of payware Emby? Probably most code would fit with copy/paste, dunno, need to install it and check.

**Comment by mastercho** [comment]
Oh, well Jelly and Emby are 100% same, not need even copy paste just rename maybe that is works for both

**Reply by Purfview** [reply]
Looks like it's not same after all, actually Emby provides much better access to the libraries, when Jellyfin is almost at the same poor Plex level for same matter. 
I tried to run Emby on Jelly's code, and it actually run without issues, just Jelly's access token was rejected, and some links are a lil bit different. At same time Jelly wouldn't run on Emby's code, because it has superior features not implemented in Jelly.

My ratings evaluating GUI/Settings/Library setup/Library access:
Plex 2/10 - Crapware, I've no words how everything is so bad there. I've no idea why people use it.
Jellyfin 6/10 - Good. Would be better than Emby if it had decent access to the libraries.
Emby 8/10 - Very good. Superb access to the libraries. GUI noticeably slower than Jellyfin.

Library tested was 5 movies + 1 series. Plex autodetected only 2 of them, when Emby/Jellyfin autodetected them all.

**Comment by Purfview** [comment]
From v13.1 if "green" then button will open actual movie/series.
